src/demo/animation.rs

```diff
@@ -24,6 +24,7 @@ pub(super) fn plugin(app: &mut App) {
                 trigger_step_sfx,
             )
                 .chain()
+                .run_if(in_state(Screen::Playing))
```

Comment:
Is this run condition necessary? Maybe you want a ducky walking around your title screen.

Reply:
Another option is something like `.run_if(resource_exists::<PlayerAssets>)`, instead of coupling to the `Screen` state.

Reply:
> .run_if(resource_exists::<PlayerAssets>)

Thats a reasonable way to do it I think. That being said, I do think tying into states for determining if a resource can/should be used is a good idea generally.